addons/website/static/src/js/theme_preview_form.js

```diff
@@ -0,0 +1,103 @@
+odoo.define('website.theme_preview_form', function (require) {
+"use strict";
+
+var FormController = require('web.FormController');
+var FormView = require('web.FormView');
+var viewRegistry = require('web.view_registry');
+var core = require('web.core');
+var _t = core._t;
+var qweb = core.qweb;
+
+var ThemePreviewController = FormController.extend({
+    events: Object.assign({}, FormController.prototype.events, {
+        'click .o_use_theme': '_onUseThemeClick',
+        'click .o_switch_theme': '_onSwitchThemeClick',
+        'click .o_switch_mode_button': '_onSwitchButtonClick',
+    }),
+
+    init: function () {
+        this._super.apply(this, arguments);
+        this.$switchButton = false;
+    },
+    // -------------------------------------------------------------------------
+    // Public
+    // -------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    renderButtons: function ($node) {
+        var $previewButton = $(qweb.render('ThemePreview.Buttons'));
+        $node.html($previewButton);
+        this.$switchButton = this._renderSwitchButton();
+        $node.find('.o_switch_mode_button').replaceWith(this.$switchButton);
+    },
+    /**
+     * @override
+     */
+    start: function () {
+        this.$el.addClass('o_view_form_theme_preview_controller');
+        return this._super.apply(this, arguments);
+    },
+
+    // -------------------------------------------------------------------------
+    // Handlers
+    // -------------------------------------------------------------------------
+    _renderSwitchButton: function () {
+        var isMobile = !!this.$('.is_mobile').length;
+        return $(qweb.render('ThemePreview.SwitchModeButton', {
+            'icon': isMobile ? 'fa-desktop' : 'fa-refresh',
+            'PreviewType': isMobile ? _t('Desktop') : _t('Mobile'),
+        }));
+    },
+    /**
+     * Handler called when user click on 'Desktop/Mobile preview' button in forw view.
+     *
+     * @private
+     */
+    _onSwitchButtonClick: function () {
+        this.$('.preview_frame').toggleClass('is_mobile');
+        const $switchButton = this.$switchButton;
+        this.$switchButton = this._renderSwitchButton();
+        $switchButton.replaceWith(this.$switchButton);
+    },
+    /**
+     * Handler called when user click on 'Choose another theme' button in forw view.
+     *
+     * @private
+     */
+    _onSwitchThemeClick: function () {
+        this.trigger_up('history_back');
+    },
+    /**
+     * Handler called when user click on 'Use this theme' button in forw view.
+     *
+     * @private
+     */
+    _onUseThemeClick: function () {
+        const $loader = $(`
+            <div class='position-fixed fixed-top fixed-left h-100 w-100 bg-white d-flex justify-content-center align-items-center'>
+                <div class='theme_install_loader'></div>
+            </div>`
```

Comment:
I'd say to use a qweb template but I guess this is simple enough to justify using jquery

Reply:
starting with a simple <div class='loader'/> ...
But you are right, at the end, the code become bigger and should be done with qweb